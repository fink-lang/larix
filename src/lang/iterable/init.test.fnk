{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'

{parse_expr} = import '../../parser.fnk'


describe 'fold item, accu: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        fold item, accu=0:
          item + accu
          # trailing comment
      '
      to_match_snapshot


describe 'unfold accu: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        unfold curr=start:
          start + accu
      '
      to_match_snapshot

  it 'parses with item, accu result', fn:
    expect
      parse_expr '
        unfold prev=0:
          (prev * 2, prev + 1)
      '
      to_match_snapshot

  it 'parses no args', fn:
    expect
      parse_expr '
        unfold:
          foo
          # trailing comment
      '
      to_match_snapshot


describe 'map item: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        map item:
          item * 2
          # trailing comment
      '
      to_match_snapshot


describe 'filter item: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        filter item:
          item % 2 == 0
          # trailing comment
      '
      to_match_snapshot


describe 'while item: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        while item:
          item % 2 == 0
          # trailing comment
      '
      to_match_snapshot


describe 'until item: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        until item:
          item % 2 == 0
      '
      to_match_snapshot


describe 'find item: ...', fn:

  it 'parses', fn:
    expect
      parse_expr '
        find item:
          item % 2 == 0
          # trailing comment
      '
      to_match_snapshot

