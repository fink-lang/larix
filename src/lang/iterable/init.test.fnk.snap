// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter item: ... parses 1`] = `
block filter (1:0-4:0)
  ident (1:7-1:11) item
  :
  comp == (2:2-2:15)
    arithm % (2:2-2:10)
      ident (2:2-2:6) item
      number (2:9-2:10) 2
    number (2:14-2:15) 0
  # trailing comment
`;

exports[`fold item, accu: ... parses 1`] = `
block fold (1:0-4:0)
  ident (1:5-1:9) item
  assign = (1:11-1:17)
    ident (1:11-1:15) accu
    number (1:16-1:17) 0
  :
  arithm + (2:2-2:13)
    ident (2:2-2:6) item
    ident (2:9-2:13) accu
  # trailing comment
`;

exports[`map item: ... parses 1`] = `
block map (1:0-4:0)
  ident (1:4-1:8) item
  :
  arithm * (2:2-2:10)
    ident (2:2-2:6) item
    number (2:9-2:10) 2
  # trailing comment
`;

exports[`unfold accu: ... parses 1`] = `
block unfold (1:0-3:0)
  assign = (1:7-1:17)
    ident (1:7-1:11) curr
    ident (1:12-1:17) start
  :
  arithm + (2:2-2:14)
    ident (2:2-2:7) start
    ident (2:10-2:14) accu
`;

exports[`unfold accu: ... parses no args 1`] = `
block unfold (1:0-4:0):
  ident (2:2-2:5) foo
  # trailing comment
`;

exports[`unfold accu: ... parses with item, accu result 1`] = `
block unfold (1:0-3:0)
  assign = (1:7-1:13)
    ident (1:7-1:11) prev
    number (1:12-1:13) 0
  :
  group (2:2-2:22)
    arithm * (2:3-2:11)
      ident (2:3-2:7) prev
      number (2:10-2:11) 2
    arithm + (2:13-2:21)
      ident (2:13-2:17) prev
      number (2:20-2:21) 1
`;

exports[`until item: ... parses 1`] = `
block until (1:0-3:0)
  ident (1:6-1:10) item
  :
  comp == (2:2-2:15)
    arithm % (2:2-2:10)
      ident (2:2-2:6) item
      number (2:9-2:10) 2
    number (2:14-2:15) 0
`;

exports[`while item: ... parses 1`] = `
block while (1:0-4:0)
  ident (1:6-1:10) item
  :
  comp == (2:2-2:15)
    arithm % (2:2-2:10)
      ident (2:2-2:6) item
      number (2:9-2:10) 2
    number (2:14-2:15) 0
  # trailing comment
`;
